doc/Type/independent-routines.pod6

```diff
@@ -1164,6 +1164,65 @@ assigning L<Empty|/type/Slip#index-entry-Empty-Empty>; for everything else,
 equivalent to assigning L<Nil|/type/Nil> or C<Empty> in the case of arrays or
 hashes, whose use is advised.
 
+=head1 Array manipulation
+
+Routines that manipulate arrays and other containers.
+
+=head2 sub pop
+
+Defined as:
+
+    proto sub pop($, *%)
+    multi sub pop(@a)
+
+Calls method L<C<pop>|/routine/pop> on the container, which is supposed to
+remove and return the last element, or fail if the container is empty.
+
+=head2 sub shift
+
+Defined as:
+
+    proto sub shift($, *%)
+    multi sub shift(@a)
+
+Calls method L<C<shift>|/routine/shift> on the container which is supposed to
+remove and return the first element, or fail if the container is empty.
+
+=head2 sub push
+
+Defined as:
+
+    proto sub push($, |)
+    multi sub push(\a, \b)
+    multi sub push(\a, **@b)
+
+Probably best thought of as a simple:
+
+    sub push(\a, **@b)
+
+Calls method L<C<push>|/routine/push> on container C<a> which is supposed to
+add C<b> or the values in C<|@b> to the end of an array. If C<b> is a C<Slip>,
+the container is supposed to add the values produced by
+L<C<b.iterator>|/routine/iterator>.
+
+=head2 sub append
+
+Defined as:
+
+    proto sub append($, |)
+    multi sub append(\a, \b)
+    multi sub append(\a, **@b)
+
+Probably best thought of as a simple:
+
+    sub append(\a, +b)
+
+Calls method L<C<append>|/routine/append> on container C<a> which is supposed to
+add C<b> or the values in C<@b> to the end of a container. In contrast with
+L<sub C<push>|#routine_push>, the container method is supposed to add the values
+produced by the L<iterator|/routine/iterator> if called with a B<single>
+non-slipping C<Iterable> that is not inside a C<Scalar> container.
+
```

Comment:
There are no examples anywhere of append as a sub, but it might be a good idea to insert them here as long as we're giving it some space.

Reply:
I agree.